src/main/java/com/aws/greengrass/clientdevices/auth/certificate/handlers/CACertificateChainChangedHandler.java

```diff
@@ -43,7 +44,10 @@ public void listen() {
      */
     @Override
     public void accept(CACertificateChainChanged event) {
-        // TODO: Move retry logic from the domain to here, based on the result returned from the useCase
-        useCases.get(RegisterCertificateAuthorityUseCase.class).apply(null);
+        try {
+            useCases.get(RegisterCertificateAuthorityUseCase.class).apply(null);
+        } catch (UseCaseException e) {
+            // TODO: Move retry logic from the domain to here, based on the result returned from the useCase
```

Comment:
better to have a failure log here until we finish this TODO

Reply:
We did not have anything before, but I agree